pinner/pinmgr.go

```diff
@@ -95,6 +97,38 @@ type PinningOperation struct {
 	MakeDeal bool
 }
 
+//TODO put this as a subfield inside PinningOperation
+type PinningOperationData struct {
+	Obj  cid.Cid
+	Name string
+	//Peers       []*peer.AddrInfo
+	Meta        string
+	Status      types.PinningStatus
+	UserId      uint
+	ContId      uint
+	Replace     uint
+	Location    string
+	SkipLimiter bool
+	MakeDeal    bool
+}
+
+func getPinningData(po *PinningOperation) PinningOperationData {
+	return PinningOperationData{
+		Obj:  po.Obj,
+		Name: po.Name,
+		//Peers:       po.Peers,
```

Comment:
the peers are needed

Reply:
thanks! That was my kind of only question - any tips on how to serialize the value for it ? 

also i'm debating whether the work queue guard is necessary or if backing the queue to disk is sufficient.

it would depend on whether there is erroneously duplicated data in the queue or not.

Also, no worries if it's too much but it would be helpful if you could you share a quick explanation of what the Peers datastructure is

Reply:
I think the `ContId` is sufficient enough to use for dups check